finat/physically_mapped.py

```diff
@@ -142,6 +181,14 @@ def jacobian_at(self, point):
         :returns: A GEM expression for the Jacobian, shape (gdim, tdim).
         """
 
+    @abstractmethod
+    def detJ_at(self, point):
+        """The determinant of the jacobian of the physical coordinates at a point.
+
+        :arg point: The point in reference space to evaluate the Jacobian determinant.
+        :returns: A GEM expression for the Jacobian determinant.
+        """
+
     @abstractmethod
     def reference_normals(self):
         """The (unit) reference cell normals for each facet.
```

Comment:
`reference_edge_tangents` is added in firedrakeproject/tsfc#225, but not here.

Reply:
we can use it in a later PR once the tsfc changes land.  We need to make sure they are scaled/not scaled appropriately for this use (FIAT supports two kinds).

Reply:
Ok, so `reference_edge_tangets` is not used here in this PR, fine.